### ISSUE TITLE
Add `type_literal_in_constant_pattern`

### ISSUE THREAD

**Issue by johnpryan** [dart-lang/lints#117]
I recently ran into https://github.com/dart-lang/language/issues/2911 and this seems like a good candidate to add to the recommended set. It's probably something users want more often than not.

**Comment by lrhn** [comment]
I'd really want this warning to be enabled by default in the analyzer, if at all possible.
Even if it's just as a "hint".

Otherwise, I'd put it in `core`. It's warning about a highly error-prone pattern, and I'd prefer knowing that packages I depend on are aware of the pitfall, and have taken steps to avoid it.

**Comment by devoncarew** [comment]
This is approved for the `core` set.

**Comment by Fernandomr88** [comment]
so I have a function that basically is this

```
get<T>() {
 switch(T) {
  case Person:
   //code
  case Dog:
   //code
  }
}
```

which is called like `get<Person>();`

and by using the notation suggested by this rule (`Person _`), it stopped working.

How am I supposed to change my code to be adequate to this linter rule?

**Comment by bwilkerson** [comment]
It sounds like the rule might need to be changed to not produce a diagnostic if the value being switched over is `Type`.
